src/scripts/outlet-registration/periods.js

```diff
@@ -0,0 +1,73 @@
+var moment = require('moment');
+
+/**
+ * Helper module to create periods of time according to the different supported periods from dhis:
+ *  'daily','weekly','monthly','yearly'
+ */
+function Periods(){
+    this.config = {
+        daily:{
+            format:'YYYY-MM-DD',
+            periodKey:'day'    
+        },
+        weekly:{
+            format:'YYYY[W]WW',
+            periodKey:'week'    
+        },
+        monthly:{
+            format:'YYYYMM',
+            periodKey:'month'    
+        },
+        yearly:{
+            format:'YYYY',
+            periodKey:'year'    
+        }                        
+    }
+}
+
+/**
+ * Builds an array of 'num' strings, where every item corresponds to the period of time formatted according to the given type.
+ * @param type Type of periods to returns 'daily','weekly','monthly','yearly'
+ * @param num Num of periods to build from now into the past
+ */
+Periods.prototype.buildPeriods = function(type,num){
+    var periods = [];
+    var periodConfig = this.config[type] || this.config[type.toLowerCase()];
+    //Error -> this should not happen
+    if(!periodConfig){
+        return periods;
+    }
+    
+    var now = moment();
+    for (var i = 0; i < num; i++) {
+        //Format moment according to type
+        var formattedNow = now.format(periodConfig.format);
+        //Add moment to array
+        periods.push(formattedNow);
+        //Move backwards
+        now = now.add(-1, periodConfig.periodKey);
+    }
+    return periods;
+}
+
```

Comment:
I would refactor this function a bit. Suggestions:
- In general, try to use one single 'return'.
- Try to avoid creating a variable if you are using once.

So for instance you can have some like this:
```
    var periods = [];
    var periodConfig = this.config[type] || this.config[type.toLowerCase()];
    //Error -> this should not happen
    if(!periodConfig){
        var now = moment();
        for (var i = 0; i < num; i++) {
            //Format moment according to type and add moment to array
            periods.push(now.format(periodConfig.format));
            //Move backwards
            now = now.add(-1, periodConfig.periodKey);
        }
    }    
    return periods;
```